docpages/example_programs/the_basics/using_callback_functions.md

```diff
@@ -0,0 +1,97 @@
+
+    bot.on_log(dpp::utility::cout_logger());
+
+    bot.on_slashcommand([&bot, gm](const dpp::slashcommand_t& event) -> void {
+        if (event.command.get_command_name() == "msg-get") {
+
+            // get a message using its own and channel's ID
+            bot.message_get(gm.message_id, gm.channel_id, [&bot, event](const dpp::confirmation_callback_t& callback) -> void {
+                if (callback.is_error()) { // catching an error to log it
+                    bot.log(dpp::loglevel::ll_error, callback.get_error().message);
+                    return;
+                }
+
+                // callback.value will contain the message we are getting but the type will be std::variant, so we will have to get the message itself.
+                auto message = callback.get <dpp::message>();
+                //std::get <dpp::message>(callback.value) would give the same result
+
+                event.reply(message); // replies with the same message
+            });
+        }
+        if (event.command.get_command_name() == "channel-create") {
+
+            //create a text channel
+            dpp::channel channel = dpp::channel()
+                .set_name("test")
+                .set_type(dpp::channel_type::CHANNEL_TEXT)
+                .set_guild_id(1112789761032200306)
+                .set_parent_id(0);
+            bot.channel_create(channel, [&bot, event](const dpp::confirmation_callback_t& callback) -> void {
+                if (callback.is_error()) { // catching an error to log it
+                    bot.log(dpp::loglevel::ll_error, callback.get_error().message);
+                }
+
+                // callback.value will contain the channel we are creating but the type will be std::variant, so we will have to get the channel itself.
+                auto channel = callback.get <dpp::channel>();
+                // std::get <dpp::channel>(callback.value) would give the same result
+
+                // reply with the created channel information
+                dpp::message message = dpp::message(fmt::format("The channel's name is {0}, ID is {1} and type is {2}", channel.name, channel.id, channel.get_type()));
+                /* note that channel types are represented as numbers */
+                event.reply(message);
+            });
+        }
+        if (event.command.get_command_name() == "msg-error") {
+            bot.message_get(69, 420, [event](const dpp::confirmation_callback_t& callback) -> void {
+
+                // the error will occur since there is no message with ID '69' that is in a channel with ID '420' (at least the bot can't see it)
+                if (callback.is_error()) {
+                    event.reply(callback.get_error().message);
+                    return;
+                }
+
+                // we won't be able to get here because of the return; statement
+                auto message = callback.get <dpp::message>();
```

Comment:
Let's not encourage people to get used to auto when it's easy to type dpp::message

Reply:
Ehhhh i disagree here
dpp::message is directly referenced as a template parameter anyways, it's clear what it is

Reply:
But i *would* remove the space between get and the template parameter

Reply:
My IDE barks at me for not typing auto but as you wish master

Reply:
I don't agree. Beginners should learn to be explicit imo, we also do not use auto in any examples so that just doesn't make sense to randomly introduce it here.

Reply:
Correction, we don't use auto in any examples apart from the ones in coro. However, I feel it's justified in the coro docs.

Reply:
auto is more (not no more get it)